### PR TITLE
[tailwindcss] Make darkMode config option optional

### PR DESCRIPTION
**Pull request by janklimo**
`darkMode` becomes optional in v3 ([upgrade guide](https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration)).

### CODE REVIEW

File: types/tailwindcss/test/tailwindcss-optional-config-tests.ts
Comment (nicolas377) [inline types/tailwindcss/test/tailwindcss-optional-config-tests.ts:1]: The relative import is whats causing the CI to fail, here's the fix (haven't tested it):
```suggestion
import type { TailwindConfig } from 'tailwindcss';
```

**Review by nicolas377 — CHANGES_REQUESTED** [review]
+1 to this change! I've been struggling with this issue a lot recently.

**Comment by janklimo** [comment]
Thanks for the ping @nicolas377. Fixed the relative import 👍

**Review by nicolas377 — APPROVED** [review]
Looks good! I don't maintain this package though, so it'll be a wait until this actually gets merged.

**Comment by boian-ivanov** [comment]
This has been exactly what I've been struggling with. Glad to see I've not been alone. Hopefully it can be merged soon.

**Comment by nicolas377** [comment]
I think we need more people to step up to be maintainers for tailwindcss, since its such a popular library, and the current maintainers seem to be occupied.

**Review by boian-ivanov — APPROVED** [review]

**Comment by janklimo** [comment]
@dolanmiu @DylanVann can you chime in? I maintain other DT packages, happy to help maintaining this moving forward.

**Review by dolanmiu — APPROVED** [review]

**Comment by janklimo** [comment]
Ready to merge

**Pull request merged**
